### ISSUE TITLE
Rendering a sequence in a media file consumes a lot of computer resources.

### ISSUE THREAD

**Issue by smolnp** [jliljebl/flowblade#554]
VKontakte message from the user - "It consumes a lot of resources .... Openshot and Kdenlive - much less ... 8Gb-RAM + 9.7-swap ... Davinchi on Windows loaded the processor so much, but DVNCH has 12Gb RAM system requirements. .. And just a couple of media files for 3 minutes in HD1080 I wanted to convert ... And that the GPU accelerator is not in "performance" is bad ... And for some reason it does not show the progress of the assembly ... Just the window is hanging ... "( Rosa linux r10, MLT v6.10.0, Flowblade 1.16).
![qyela1lf2ta](https://user-images.githubusercontent.com/26217756/46745831-81e23580-ccb6-11e8-931a-f78cf4342839.jpg)
Checked on Xubuntu 18.04 MLT v6.6 Flowblade 1.16. Result on the screenshot:
![_2018-10-10_18-06-24](https://user-images.githubusercontent.com/26217756/46746606-0a150a80-ccb8-11e8-90c7-a2b26e388cbe.png)

**Comment by jliljebl** [comment]
Thanks for providing additional info, seems that Issue is that rendering is using 100% on all CPUs. That is sort of the desired functionality when rendering because then the render will be done as quickly as possible, but if we're getting different behavior from KDEnlive this should be looked into. 

This is (probably) not the same as #327 so I'll keep open as a separate issue.

**Comment by jliljebl** [comment]
Ok, there is a clear difference on CPU usage between KDElie and Flowblade, this needs to be more investigation.

**Comment by jliljebl** [comment]
This issue is not present "Apple Pro Re" rendering and is less severe in other codecs. Maybe just giving different render args will affect this.

I will return to this before release, but on one thing that seems to be in play is that Flowblade does not use the "preview" MLT consumer, and does use more processor power in some situations while editing, but lets see if there are some parameter changes that we can do to affect processor usage.

**Comment by jliljebl** [comment]
So far only thing I found that could be related are two threads complaining that rendering **does not use all CPUs to max**, the opposite of what is happening here. These suggest that OpenCV is somehow messing with threads, quote: 

"When MLT initializes, it checks to see what all frei0r plugins exists, and simply loading an OpenCV-based plugin may initialize something in OpenCV without that plugin being used in a composition"

https://github.com/mltframework/mlt/issues/274
https://bugzilla.novell.com/show_bug.cgi?id=1068792

It could be that Flowblade is doing the right thing as opposed to programs.

I did some tests on a project of mine:

codec                                Render Time                 CPU use
-----------------------------------------------------------------------------------------------------------
h264 / mp4                        12m                               4x CPU 100%

Theora                               15m                               1 x CPU 100%, 3 x CPU 15%   

Lossles mp4                       7m                                4 x CPU 85%  

MPEG2                               7m                               4 x CPU 90%

Apple Prores                       14m                             1 x CPU 100, 3 x CPU 15%


So h264 / mp4  was not getting much for 4x CPU usage, Lossless mp4 and MPEG2 may also just have been faster to render, so I'm not sure they got the benefit from increased CPU usage. Memory use also kept increasing when running these in Batch Render tool.

We will need to keep investigating in next cycle until we have enough info to make conclusions.
